x-pack/platform/plugins/shared/fleet/common/types/models/cloud_connector.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { PackagePolicyConfigRecordEntry } from './package_policy';
+
+export type CloudProvider = 'aws' | 'azure' | 'gcp';
+
+export interface CloudConnectorSecretVarValue {
+  isSecretRef: boolean;
+  id: string;
+}
+
+export interface CloudConnectorSecretVar {
+  type: 'password';
+  value: CloudConnectorSecretVarValue;
+  frozen?: boolean;
+}
+
+export interface CloudConnectorVars {
+  // AWS Role ARN variables
+  'aws.role_arn'?: PackagePolicyConfigRecordEntry | string;
+  role_arn?: PackagePolicyConfigRecordEntry | string;
+  // AWS credentials variables
+  'aws.credentials.external_id'?: CloudConnectorSecretVar;
+  external_id?: CloudConnectorSecretVar;
+  // Azure variables
+  client_id?: CloudConnectorSecretVar;
+  tenant_id?: CloudConnectorSecretVar;
+}
+
+export interface CloudConnectorSO {
+  id: string;
+  name: string;
+  cloudProvider: CloudProvider;
+  vars: CloudConnectorVars;
```

Comment:
I feel that these typings could be made more strict by having `vars` definition depend on which `cloudProvider`.

you could probably have separate typings like `AwsCloudConnectorVars`, `AzureCloudConnectorVars`, etc

this way we don't have to treat all the var properties as optional